### PR TITLE
ICU-22723 CLDR 46 46 beta3 to main

### PR DESCRIPTION
**Pull request by DraganBesevic**
TODO: Please describe your changes here.

TODO: Delete the following lines but _keep and fill out_ (!) the checklist below.

Thank you for your pull request!\
General info on contributing: Please see https://github.com/unicode-org/icu/blob/main/CONTRIBUTING.md

Contributors license agreement (CLA):
- You will be automatically asked to sign the CLA before the PR is accepted.
- To sign the CLA: https://cla-assistant.io/unicode-org/icu

For terms of use and license, see https://www.unicode.org/terms_of_use.html

TODO: End of lines to delete.

#### Checklist
- [ ] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22723 
  - For minor changes you can use one of the following per-release tickets:
  - ICU 77 code warnings/version updates: ICU-22920 — Fix compiler warnings. Update versions of code-related dependencies (e.g., dependabot).
  - ICU 77 docs minor fixes: ICU-22921 — User Guide & API docs typos etc., and version updates (e.g., dependabot for User Guide)
- [ ] Required: The PR title must be prefixed with a JIRA Issue number. Example: "ICU-1234 Fix xyz"
- [ ] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. Example: "ICU-1234 Fix xyz"
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

Beta 3 integration for CLDR 46.

Includes updates for tzdata to 2024b and adding new measurement units (duration-night and light-speed).

Known issues:

1. There is still a problem with per billion unit (a.k.a 1e9). All references are temporarily removed until ICU supports handling this unit: https://unicode-org.atlassian.net/browse/ICU-22781

2. "metric-ton" was renamed to "tonne", while keeping its internal value within the code the same (METRIC_TON). That confused the tool which automatically creates APIs for units. The tool doesn't create createMetricTon()/getMetricTon() any more. Now it creates createTonne()/getTonne(). I hacked the tool to manually create those existing APIs, there is possible improvement there for the future

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by mihnita — APPROVED** [review]
LGTM.

Thank you very much.
Most of it looks familiar from my struggles yesterday (all the resources).
So it wasn't that hard to check.

Mihai

**Review by pedberg-icu — APPROVED** [review]

**Comment by DraganBesevic** [comment]
> LGTM.
> 
> Thank you very much. Most of it looks familiar from my struggles yesterday (all the resources). So it wasn't that hard to check.
> 
> Mihai

Thank YOU Mihai.

Without you pulling all the changes for per billion, I wouldn't be able to wrap this up

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java [hunk 1]
```diff
@@ -4129,11 +4128,46 @@ static void generateCXXHConstants(String thisVersion) {
...
+    // Add the headers for "metric-ton"
+    // The tool won't create them any more    
+    private static void addCXXHForMetricTon() {
+        System.out.println("    /**");
+        System.out.println("     * Returns by pointer, unit of mass: metric-ton");
+        System.out.println("     * (renamed to tonne in CLDR 42 / ICU 72).");
+        System.out.println("     * Caller owns returned value and must free it.");
+        System.out.println("     * Note: In ICU 74 this will be deprecated in favor of");
+        System.out.println("     * createTonne(), which is currently draft but will");
+        System.out.println("     * become stable in ICU 74, and which uses the preferred naming.");
```
Comment (markusicu) [inline icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java:4151]: @DraganBesevic in a follow-up PR please fix these docs (here and the next function), as discussed.

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java [hunk 2]
```diff
@@ -4129,11 +4128,46 @@ static void generateCXXHConstants(String thisVersion) {
                     System.out.println("#endif /* U_HIDE_DRAFT_API */");
                 }
                 System.out.println("");
+                // Hack: METRIC-TON unit changed its name from "metric-ton" to "tonne"
+                // In order to preserve the existing APIs for "metric-ton" we need to
+                // add those APIs manually
+                if (name.equals("Tonne")) {
+                    addCXXHForMetricTon();
```
Comment (markusicu) [inline icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/MeasureUnitTest.java:4135]: Why is there nothing like this for Java?

**Review by markusicu — COMMENTED** [review]
Thank you!!

**Pull request merged**